src/Inference.ts

```diff
@@ -124,13 +128,25 @@ export class TypeInference {
                     unionBuilder.addArray(cjson.getArrayForValue(value));
                     break;
                 case Tag.Date:
-                    unionBuilder.addStringType("date");
+                    if (this._inferDates) {
+                        unionBuilder.addStringType("date");
+                    } else {
```

Comment:
This is not the way to do it.  We already have the string type mapping, which specifies which string types map to which.  That’s how, for example, in C# `data` and `time` maps to `date-time`, and for all other languages all date/time types map to `string`.  All you have to do is use a string type mapping that maps everything to `string`.

Reply:
Ah ok, so I just need to pass the date inference flag into the target language.

Reply:
I’ll have to take a look at the code. I’d rather not bother every target language with it.

Reply:
But then this has to be repeated for each target language? I want an easy way to turn off date inference for any language.

Reply:
That's what I mean - I want to it to work across all target languages without having to put code into each target language.

Reply:
Here's where you can override the string type mapping without meddling with target languages:

https://github.com/quicktype/quicktype/blob/master/src/index.ts#L134